File: Android/java/season1/app/src/main/java/com/kplo/samplesenti/login/MainActivity.java
package com.kplo.samplesenti.login;

import android.content.Intent;
import android.support.design.widget.TextInputEditText;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.util.Log;
import android.view.View;
import android.widget.RelativeLayout;

import com.kplo.samplesenti.R;

public class MainActivity extends AppCompatActivity {

    TextInputEditText TextInputEditText_email, TextInputEditText_password;
    RelativeLayout RelativeLayout_login;
    String emailOK = "123@gmail.com";
    String passwordOK = "1234";

    String inputEmail = "";
    String inputPassword = "";
    //{"key":"value","key":"value"}
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);


        TextInputEditText_email     = findViewById(R.id.TextInputEditText_email);
        TextInputEditText_password  = findViewById(R.id.TextInputEditText_password);
        RelativeLayout_login        = findViewById(R.id.RelativeLayout_login);

        //1. 값을 가져온다 - 검사 (123@gmail.com / 1234)
        //2. 클릭을 감지한다
        //3. 1번의 값을 다음 액티비티로 넘긴다
        RelativeLayout_login.setClickable(false);
        TextInputEditText_email.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {

            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {

                if(s != null) {
                    Log.d("SENTI", s.toString());
                    inputEmail = s.toString();
                    RelativeLayout_login.setEnabled(validation());
                }


            }

            @Override
            public void afterTextChanged(Editable s) {

            }
        });

        TextInputEditText_password.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {

            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {

                if(s != null) {
                    Log.d("SENTI", s.toString());
                    inputPassword = s.toString();
                    RelativeLayout_login.setEnabled(validation());
                }

            }

            @Override
            public void afterTextChanged(Editable s) {

            }
        });

//        RelativeLayout_login.setClickable(true);
        RelativeLayout_login.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                Log.d("SENTI", "onClick");

                String email = TextInputEditText_email.getText().toString();
                String password = TextInputEditText_password.getText().toString();

                Intent intent = new Intent(MainActivity.this, LoginResultActivity.class);
                intent.putExtra("email", email);
                intent.putExtra("password", password);
                startActivity(intent);
            }
        });
    }

    public boolean validation() {
        Log.d("SENTI", inputEmail + " / " + inputPassword + " / " + (inputEmail.equals(emailOK)) + " / " + (inputPassword.equals(passwordOK)));
        return inputEmail.equals(emailOK) &&  inputPassword.equals(passwordOK);
    }
}

File: Android/kotlin/FutureTask/app/src/main/java/com/sentilab/futuretask/MainActivity.kt
package com.sentilab.futuretask

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import java.util.concurrent.ExecutorService
import java.util.concurrent.Executors
import java.util.concurrent.FutureTask

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        futureTask()
    }

    fun futureTask() {
        //1. FutureTask
        val futureTaskCallable = FutureTask {

            return@FutureTask "RESULT :: futureTaskCallable"
        }
        val futureTaskRunnable = FutureTask ({

        }, "RESULT :: futureTaskRunnable")

        //2. ExecutorService
        val executorService = Executors.newFixedThreadPool(2)
        executorService.execute(futureTaskCallable)
        executorService.execute(futureTaskRunnable)

        //+ result
        Log.d("SENTI", "SENTI ::: " + futureTaskCallable.get())
        Log.d("SENTI", "SENTI ::: " + futureTaskRunnable.get())

        //+++ for loop
    }
}